pkg/cli/alpha/config-gen/types.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configgen
+
+import (
+	"io/ioutil"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+)
+
+// KubebuilderConfigGen implements the API for generating configuration
+type KubebuilderConfigGen struct {
+	metav1.TypeMeta   `json:",inline" yaml:",omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
```

Comment:
Is there any requirement for them to be a k8s object-like type or just a design decission?

Reply:
A design decision, but consistent with other tools such as kustomize.

Reply:
Kustomize plugins also require this format